README.md

```diff
@@ -224,6 +207,56 @@ The component also has the ability to track the status of a job which was pushed
 
 For more details see [the guide](docs/guide/README.md).
 
+## Middleware pipelines
+
+Any message pushed to a queue or consumed from it passes through two different middleware pipelines: one pipeline
+on message push and another - on message consume. The process is the same as for the HTTP request, but it is executed
+twice for a queue message. That means you can add extra functionality on message pushing and consuming with configuration
+of the two classes: `PushMiddlewareDispatcher` and `ConsumeMiddlewareDispatcher` respectively.
+
+You can use any of these formats to define a middleware:
+- A ready-to-use middleware object: `new FooMiddleware()`. It must implement either `MiddlewarePushInterface`,
+    or `MiddlewareConsumeInterface` depending on the place you use it.
+- An array in the format of [yiisoft/definitions](https://github.com/yiisoft/definitions).
+    **Only if you use yiisoft/definitions and yiisoft/di**.
+- A `callable`: `fn() => // do stuff`, `$object->foo(...)`, etc. It will be executed through the [yiisoft/injector](yiisoft/injector), so all the dependencies of your callable will be resolved
+- A string for your DI container to resolve the middleware, e.g. `FooMiddleware::class`
+
+Middleware will be executed forwards in the same order they are defined. If you define it like the following:
+`[$middleware1, $midleware2]`, the execution will look like this:
+1. `$middleware1`
+1. `$middleware2`
+1. push to a queue server *OR* handle on consume
+1. `$middleware2`
+1. `$middleware1`
```

Comment:
```suggestion
```mermaid
graph LR
    StartPush((Start)) --> PushMiddleware1[$middleware1] --> PushMiddleware2[$middleware2] --> Push(Push to a queue)
    -.-> PushMiddleware2[$middleware2] -.-> PushMiddleware1[$middleware1]
    PushMiddleware1[$middleware1] -.-> EndPush((End))
    

    StartConsume((Start)) --> ConsumeMiddleware1[$middleware1] --> ConsumeMiddleware2[$middleware2] --> Consume(Consume / handle)
    -.-> ConsumeMiddleware2[$middleware2] -.-> ConsumeMiddleware1[$middleware1]
    ConsumeMiddleware1[$middleware1] -.-> EndConsume((End))
```
```

Reply:
That would be:

```mermaid
graph LR
    StartPush((Start)) --> PushMiddleware1[$middleware1] --> PushMiddleware2[$middleware2] --> Push(Push to a queue)
    -.-> PushMiddleware2[$middleware2] -.-> PushMiddleware1[$middleware1]
    PushMiddleware1[$middleware1] -.-> EndPush((End))
    
    StartConsume((Start)) --> ConsumeMiddleware1[$middleware1] --> ConsumeMiddleware2[$middleware2] --> Consume(Consume / handle)
    -.-> ConsumeMiddleware2[$middleware2] -.-> ConsumeMiddleware1[$middleware1]
    ConsumeMiddleware1[$middleware1] -.-> EndConsume((End))
```